build/optaplanner-build-parent/pom.xml

```diff
@@ -26,7 +26,10 @@
     <version.ch.qos.logback>1.2.9</version.ch.qos.logback>
     <version.org.apache.logging.log4j>2.17.2</version.org.apache.logging.log4j>
     <version.com.thoughtworks.xstream>1.4.19</version.com.thoughtworks.xstream>
+    <version.io.javaoperatorsdk>3.0.0.RC1</version.io.javaoperatorsdk>
```

Comment:
Unfortunately, the "dependent resources" feature is unavailable in the final version. Not using version 3 means introducing a lot of boilerplate code to get the same functionality.
Meanwhile, RC4 has become available that I will upgrade to so that we keep as close to the following final version as possible.

Reply:
As there is no guarantee the javaoperatorsdk final version gets released before our next release, we cannot merge this PR yet.

Reply:
In the end, I am excluding the `optaplanner-operator` from the default build in order to merge this PR.